includes/blocks/class-wc-block-library.php

```diff
@@ -0,0 +1,254 @@
+				'style'           => 'wc-block-style',
+			)
+		);
+	}
+
+	/**
+	 * Adds a WooCommerce category to the block inserter.
+	 *
+	 * @param array $categories Array of categories.
+	 * @return array Array of block categories.
+	 */
+	public static function add_block_category( $categories ) {
+		return array_merge(
+			$categories,
+			array(
+				array(
+					'slug'  => 'woocommerce',
+					'title' => __( 'WooCommerce', 'woocommerce' ),
+					'icon'  => 'woocommerce',
+				),
+			)
+		);
+	}
+
+	/**
+	 * Output the wcSettings global before printing any script tags.
+	 */
+	public static function print_script_settings() {
+		global $wp_locale;
+		$code = get_woocommerce_currency();
+		// NOTE: wcSettings is not used directly, it's only for @woocommerce/components
+		//
+		// Settings and variables can be passed here for access in the app.
+		// Will need `wcAdminAssetUrl` if the ImageAsset component is used.
+		// Will need `dataEndpoints.countries` if Search component is used with 'country' type.
+		// Will need `orderStatuses` if the OrderStatus component is used.
+		// Deliberately excluding: `embedBreadcrumbs`, `trackingEnabled`.
+		$settings = array(
+			'adminUrl'      => admin_url(),
+			'wcAssetUrl'    => plugins_url( 'assets/', WC_PLUGIN_FILE ),
+			'siteLocale'    => esc_attr( get_bloginfo( 'language' ) ),
+			'currency'      => array(
+				'code'      => $code,
+				'precision' => wc_get_price_decimals(),
+				'symbol'    => get_woocommerce_currency_symbol( $code ),
+				'position'  => get_option( 'woocommerce_currency_pos' ),
+			),
+			'stockStatuses' => wc_get_product_stock_status_options(),
+			'siteTitle'     => get_bloginfo( 'name' ),
+			'dataEndpoints' => array(),
+			'l10n'          => array(
+				'userLocale'    => get_user_locale(),
+				'weekdaysShort' => array_values( $wp_locale->weekday_abbrev ),
+			),
+		);
+
+		// Global settings used in each block.
+		$block_settings = array(
+			'min_columns'       => wc_get_theme_support( 'product_grid::min_columns', 1 ),
+			'max_columns'       => wc_get_theme_support( 'product_grid::max_columns', 6 ),
+			'default_columns'   => wc_get_default_products_per_row(),
+			'min_rows'          => wc_get_theme_support( 'product_grid::min_rows', 1 ),
+			'max_rows'          => wc_get_theme_support( 'product_grid::max_rows', 6 ),
+			'default_rows'      => wc_get_default_product_rows_per_page(),
+			'placeholderImgSrc' => wc_placeholder_img_src(),
+			'min_height'        => wc_get_theme_support( 'featured_block::min_height', 500 ),
+			'default_height'    => wc_get_theme_support( 'featured_block::default_height', 500 ),
+		);
+		?>
+		<script type="text/javascript">
+			<?php // Use the wcSettings from wc-admin if already present. ?>
+			var wcSettings = wcSettings || <?php echo wp_json_encode( $settings ); ?>;
+			var wc_product_block_data = <?php echo wp_json_encode( $block_settings ); ?>;
+		</script>
+		<?php
+	}
+}
+
+if ( function_exists( 'register_block_type' ) ) {
```

Comment:
Can we move this into a `init` method?
So will make easy to make it a singleton or something like when we start implement https://github.com/woocommerce/woocommerce/pull/22849

Reply:
If we do that, will we still be able to remove the hooks [with `remove_action` in the feature plugin?](https://github.com/woocommerce/woocommerce-gutenberg-products-block/blob/5daae873583573968ece147588f91ef860d793da/woocommerce-gutenberg-products-block.php#L29-L32) I started with a singleton implementation but was having trouble getting that to work (which is needed so we can override each hook in the feature plugin)

Reply:
Since it's all static methods, should be fine to remove any action later.